src/CommunityToolkit.Maui/Behaviors/Validators/EmailValidationBehavior.shared.cs

```diff
@@ -11,29 +13,70 @@ namespace CommunityToolkit.Maui.Behaviors;
 /// </summary>
 public partial class EmailValidationBehavior : TextValidationBehavior
 {
-	/// <inheritdoc /> 
-	protected override async ValueTask<bool> ValidateAsync(string? value, CancellationToken token)
+	/// <summary>
+	/// A <see cref="Regex"/> to match an input is a valid email address
+	/// Generated by <see cref="GeneratedRegexAttribute"/>
+	/// </summary>
+	/// <returns>Generated <see cref="Regex"/></returns>
+	[GeneratedRegex(@"^[^@\s]+@[^@\s]+\.[^@\s]+$", RegexOptions.IgnoreCase, 250)]
+	protected static partial Regex EmailRegex();
+
+	/// <summary>
+	/// A <see cref="Regex"/> to match the domain of an email address
+	/// Generated by <see cref="GeneratedRegexAttribute"/>
+	/// </summary>
+	/// <returns>Generated <see cref="Regex"/></returns>
+	[GeneratedRegex(@"(@)(.+)$", RegexOptions.None, 250)]
+	protected static partial Regex EmailDomainRegex();
+
+	/// <summary>
+	/// Examines the email address domain and normalizes it to ASCII
+	/// </summary>
+	/// <param name="match"></param>
+	/// <returns></returns>
+	/// <exception cref="ArgumentException"></exception>
+	private protected static string DomainMapper(Match match)
 	{
-		if (string.IsNullOrWhiteSpace(value))
+		var domainName = match.Groups[2].Value;
+		if (domainName.StartsWith('-'))
```

Comment:
Why test _just_ for hyphen here? Afaik there are other rules on domain naming too. In my opinion it could just throw the invalid IP4/IP6 exception